src/main/java/com/google/devtools/build/lib/packages/SkylarkNativeModule.java

```diff
@@ -118,7 +119,18 @@
       throw new EvalException(loc, "illegal argument in call to glob", e);
     }
 
-    return StarlarkList.copyOf(thread.mutability(), matches);
+    String[] result = new String[matches.size()];
+    for (int i = 0; i < result.length; i++) {
+      String match = matches.get(i);
+      if (match.charAt(0) == '@') {
+        // Add explicit colon to disambiguate from external repository.
+        match = ":" + match;
+      }
+      result[i] = match;
+    }
+    Arrays.sort(result);
+
+    return StarlarkList.wrap(thread.mutability(), result);
```

Comment:
Nit: How about instead doing

  `StarlarkList.copyOf(thread.mutability(), Arrays.asList(matches))`

This way you don't need to make #wrap public. I don't think the Starlark folks want that one to be public. In particular, @alandonovan has been trying to clean up java-land visibility and API issues in the Starlark implementation.

Reply:
This is all because I wanted to avoid a unnecessary copy. This is a situation where it's definitely okay to "donate" the array to `StarlarkList`.

A possible alternative to this would be having public `StarkList` APIs to preallocate a list with the correct size and provide some way to sort it.

Possibly also I just shouldn't worry about it.

Reply:
> This is all because I wanted to avoid a unnecessary copy. 

I initially thought the same and then was going to suggest implement Iterable wrapper around an Array, and then I remembered Arrays.asList: https://docs.oracle.com/javase/9/docs/api/java/util/Arrays.html#asList-T...-

Reply:
But `matches` is a `List` not an array.

Reply:
Gosh I'm sorry -- bad copy/paste on my part.

Should be `StarlarkList.copyOf(thread.mutability(), Arrays.asList(result))`; that'd work, right?

Reply:
It adds a copy because `StarlarkList.copyOf` will call `Iterables.toArray` on it, which will make a copy.

Reply:
Okay, now I'm embarrassed :/

Let's just keep what you have right now (i.e. public #wrap) and someone on the Starlark team can take a look some other time.

Reply:
Please do not make wrap public. The arrays referenced by Tuple and StarlarkList must be completely encapsulated, like ImmutableList. You must incur one copy; copyOf(Arrays.asList(x)) is optimal.